src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -671,17 +669,16 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                 return;
             }
 
-            let mut trait_type = trait_ref.self_ty();
+            let mut suggested_ty = trait_ref.self_ty();
```

Comment:
I'm a bit confused how the types work out here. This is a `Binders<Ty>`, right?

Reply:
It's a plain `Ty` if that helps.

Reply:
Yes, I noticed that recently. Hiding the `skip_binder` call like that is...just so wrong. But it explains how this code *type-checks*, anyway.